nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-designer/ui/canvas/items/flow/import-from-registry/import-from-registry.component.ts

```diff
@@ -0,0 +1,337 @@
+        MatTableModule
+    ],
+    templateUrl: './import-from-registry.component.html',
+    styleUrls: ['./import-from-registry.component.scss']
+})
+export class ImportFromRegistry implements OnInit {
+    @Input() getBuckets!: (registryId: string) => Observable<BucketEntity[]>;
+    @Input() getFlows!: (registryId: string, bucketId: string) => Observable<VersionedFlowEntity[]>;
+    @Input() getFlowVersions!: (
+        registryId: string,
+        bucketId: string,
+        flowId: string
+    ) => Observable<VersionedFlowSnapshotMetadataEntity[]>;
+
+    saving$ = this.store.select(selectSaving);
+    timeOffset = 0;
+
+    protected readonly TextTip = TextTip;
+
+    importFromRegistryForm: FormGroup;
+    registryClientOptions: SelectOption[] = [];
+    bucketOptions: SelectOption[] = [];
+    flowOptions: SelectOption[] = [];
+
+    flowLookup: Map<string, VersionedFlow> = new Map<string, VersionedFlow>();
+    selectedFlowDescription: string | undefined;
+
+    sort: Sort = {
+        active: 'version',
+        direction: 'desc'
+    };
+    displayedColumns: string[] = ['version', 'created', 'comments'];
+    dataSource: MatTableDataSource<VersionedFlowSnapshotMetadata> =
+        new MatTableDataSource<VersionedFlowSnapshotMetadata>();
+    selectedFlowVersion: number | null = null;
+
+    constructor(
+        @Inject(MAT_DIALOG_DATA) private dialogRequest: ImportFromRegistryDialogRequest,
+        private formBuilder: FormBuilder,
+        private store: Store<CanvasState>,
+        private nifiCommon: NiFiCommon,
+        private client: Client
+    ) {
+        this.store
+            .select(selectTimeOffset)
+            .pipe(isDefinedAndNotNull(), takeUntilDestroyed())
+            .subscribe((timeOffset: number) => {
+                this.timeOffset = timeOffset;
+            });
+
+        const sortedRegistries = dialogRequest.registryClients.slice().sort((a, b) => {
+            return this.nifiCommon.compareString(a.component.name, b.component.name);
+        });
+
+        sortedRegistries.forEach((registryClient: RegistryClientEntity) => {
+            if (registryClient.permissions.canRead) {
+                this.registryClientOptions.push({
+                    text: registryClient.component.name,
+                    value: registryClient.id,
+                    description: registryClient.component.description
+                });
+            }
+        });
+
+        this.importFromRegistryForm = this.formBuilder.group({
+            registry: new FormControl(this.registryClientOptions[0].value, Validators.required),
+            bucket: new FormControl(null, Validators.required),
+            flow: new FormControl(null, Validators.required),
+            keepParameterContexts: new FormControl(true, Validators.required)
+        });
+    }
+
+    ngOnInit(): void {
+        const selectedRegistryId = this.importFromRegistryForm.get('registry')?.value;
+
+        if (selectedRegistryId) {
```

Comment:
Could we just disable the ability to import from a registry if none are available?